integration_tests/src/main/python/date_time_test.py

```diff
@@ -41,6 +41,16 @@ def test_timeadd(data_gen):
         lambda spark: unary_op_df(spark, TimestampGen(start=datetime(5, 1, 1, tzinfo=timezone.utc), end=datetime(15, 1, 1, tzinfo=timezone.utc)), seed=1)
             .selectExpr("a + (interval {} days {} seconds)".format(days, seconds)))
 
+def test_timeadd_daytime_column():
```

Comment:
Do we need to restrict this to Spark 3.3+?

Reply:
I missed the "skipif is_before_spark_330" annotation, updated.